src/edu/ucsb/nceas/metacat/properties/PropertiesWrapper.java

```diff
@@ -281,6 +301,25 @@ protected Path getMainMetadataFilePath() {
         return mainMetadataFilePath;
     }
 
+    /**
+     * The properties on the dataONE Setting class aren't synchronized with changes to the Metacat
+     * properties files. This method synchronizes (reloads) the properties' changes to the Settings
+     * class, and should be called whenever the property files are modified.
+     * NOTE that secrets passed as environment variables are accessible only by code that calls
+     *  PropertyService.getProperty(). Calling syncToSettings() does NOT make them accessible via
+     *  org.dataone.configuration.Settings
+     * @throws GeneralPropertyException if there's a problem calling Settings.augmentConfiguration()
+     */
+    protected void syncToSettings() throws GeneralPropertyException {
+        try {
+            Settings.getConfiguration();
+            Settings.augmentConfiguration(this.getDefaultPropertiesFilePath().toString());
+            Settings.augmentConfiguration(this.getSitePropertiesFilePath().toString());
+        } catch (ConfigurationException e) {
+            logAndThrow("PropertiesWrapper.syncToSettings(): augmenting DataONE settings: ", e);
+        }
+    }
+
```

Comment:
To my understanding, those environmental variables wouldn't be stored in the site properties file. So it seems their values will not be propagated to settings. Do we need code to set the Settings for those environmental variables?

Reply:
This was my question that we discussed in our backend meeting yesterday. The problem with passing these values to d1 Settings is that the current code only accepts a file path string, and reads the settings from a file on disk, which is the security issue we're trying to avoid by using env variables.

From our discussion, it sounded like none of the secrets are currently needed in the d1 code, so there is no real need to pass the vars there, yet. I'm therefore proposing waiting until we need to do this (if we ever do), before adding that complexity.

I added a note in the javadoc for this method (see line 308, above) and also in [metacat.properties on line 55](https://github.com/NCEAS/metacat/pull/1636/files/3a4005c783df382c7bf5c0326edb1fc6baa97e53#diff-3c44a421a363e2e43cea038a031a838831a483d6b78218b009bd7751b293e393R55), so other devs know the values aren't passed through to d1.  

Are you OK with this approach?

Reply:
Yeah, I forgot that. But in the `Setting` class, there is a method `getConfiguration()` and it will return the object `configuration` which holds the property values. We may call the method `setProperty` to add the new key/value pairs. But I am not sure if the setProperty will store the key/value pair into a file. If it does, I don't where it stores. Here is the link of documentation of the [method](https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/AbstractConfiguration.html#setProperty(java.lang.String,java.lang.Object)).
Your solution seems fine to me too. The above is only my thought.

Reply:
added implementation notes as guidance for our future selves, if this becomes necessary